### PR TITLE
feat(ui-concerto): Improve rendering for relationship values

### PR DESCRIPTION
**Pull request by mttrbrts**
#  Summary
This PR improves the formatting for relationship fields to better support relationship URIs (e.g. `resource:test.AnAsset#49253`). The internal URI format is difficult to understand for a novice user and the current field doesn't enforce the URI format when editing, this can easily lead to mistakes.

### Changes

- Relationship fields and relationship array fields now use the new `ConcertoRelationship` component which understands the relationship URI. This is instead of the `ConcertoInput` component which is used elsewhere.
- The new `ConcertoRelationship` component now hides the URI protocol and namespace from the path. The Type is shown in a label, and only the identifier value is editable.
- Also, this PR changes the styling of the array Add and Remove buttons. These buttons also include aria labels to improve accessibility.

### Flags
- Because `concerto` doesn't allow empty identifiers (https://github.com/accordproject/concerto/blob/f2d9942d1bda59797216468404d21e545d4d1e07/packages/concerto-core/lib/model/resourceid.js#L48) and because empty string is falsy in JavaScript, the behavior when you try to delete the last character is a little odd. If the value is empty, we have to fall back to using a non-empty default value (I've chosen `resource1`).

### Screenshots or Video
image

### Author Checklist
- [ ] Vital features and changes captured in unit and/or integration tests
- [x] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [ ] ~Extend the documentation, if necessary~
- [x] Merging to `master` from `fork:branchname`
- [x] Manual accessibility test performed
    - [x] Keyboard-only access, including forms
    - [x] Contrast at least WCAG Level A
    - [x] Appropriate labels, alt text, and instructions

### CODE REVIEW

File: packages/ui-concerto/src/lib/formgenerator.js [hunk 1]
```diff
@@ -76,10 +76,13 @@ class FormGenerator {
    */
   async loadFromText(texts) {
     this.loaded = false;
-
     this.modelManager.clearModelFiles();
     try {
-      texts.forEach(text => this.modelManager.addModelFile(text, null, true));
+      texts.forEach(text => {
+        // Unescape URL encoded relationship symbols
```
Comment (dselman) [inline packages/ui-concerto/src/lib/formgenerator.js:82]: I don't understand this. How did the model files get encoded?

File: packages/ui-concerto/package.json [hunk 2]
```diff
@@ -3,24 +3,26 @@
   "version": "0.94.0",
   "private": false,
   "dependencies": {
+    "@accordproject/concerto-core": "^0.82.8",
     "@babel/runtime": "^7.10.3",
+    "lodash.get": "^4.4.2",
```
Comment (dselman) [inline packages/ui-concerto/package.json:8]: Are these used?

File: packages/ui-concerto/src/lib/formgenerator.js (hunk 1 above)
Comment (mttrbrts) [reply packages/ui-concerto/src/lib/formgenerator.js:82]: This appears to be related to use of storyboard. When I change the values in the knob editor, I see these encoded values.

**Comment by Michael-Grover** [comment]
What does the arrow and tag mean? I don't think I've seen this in a form UI before.

What did this UI look like before the change?

image

**Comment by mttrbrts** [comment]
> What does the arrow and tag mean? I don't think I've seen this in a form UI before.
> 
> What did this UI look like before the change?
> 
> image

Previously this just appeared as a standard input field. I'm trying to find a way to distinguish between values and relationships (https://docs.accordproject.org/docs/model-relationships.html).
This is required because without it we end up displaying ugly values in the field, e.g. `resource:test.AnAsset#49253` when the only thing that I should be able to change is `49253`


In the latest iteration, I've added a popup to give more explanation
image

File: packages/ui-concerto/package.json (hunk 2 above)
Comment (mttrbrts) [reply packages/ui-concerto/package.json:8]: Yes, they are all used. In `reactformvisitor.js` and `concertoform.js`. Adding the new `urijs` dependency caused the order of dependencies to change.

**Comment by mttrbrts** [comment]
> 1. The right arrow doesn't render on Safari.
> 2. The event that is getting fired doesn't seem to include the ID for the relationship
> 
> image

I've fixed 2, I believe that this was an instance of bad URI parsing. I'm now using the concerto Relationship.toURI/fromURI functions.

1. Appears to be a preexisting issue related to rollup. I can only replicate in netlify.

**Comment by jeromesimeon** [comment]
As a developer, I really love the arrow and resource syntax under the field. I think it looks very nice.

File: packages/storybook/package.json [hunk 3]
```diff
@@ -58,6 +58,7 @@
   "scripts": {
     "storybook": "start-storybook -p 9009 -s public",
     "build-storybook": "NODE_OPTIONS='--max-old-space-size=4096' build-storybook -s public --quiet",
+    "build-storybook:watch": "NODE_OPTIONS='--max-old-space-size=4096' build-storybook -w -s public --quiet",
```
Comment (jeromesimeon) [inline packages/storybook/package.json:61]: The `README` says to use `npm run storybook`.. does this do something different?

Comment (mttrbrts) [reply packages/storybook/package.json:61]: Yes, this is helpful when developing to have the storybook auto-rebuild when the code changes. The README is correct, but personally I prefer to work like this.

**Pull request merged**